### DISCUSSION TITLE
Retaining Mesh After FW Flash

### DISCUSSION THREAD

**Discussion by embraceentropy** [Jyers/Marlin#877] [Ideas]
I have come to enjoy staying current with FW changes and trying new things. However, I've come to hate having to go through manual mesh generation every time I flash a new FW and reset/re-save settings. I've noticed things like LCD Beeper setting, UI settings, etc appear to be retained after resetting EEPROM after FW flash. 

Would it be possible to have an option to retain mesh after EEPROM reset?

**Comment by gothcha** [comment]
The UBL mesh should be retained after flashing, it certainly does for me using 1.3.2. It is stored in a separate area of memory. Make sure the mesh is saved after creation, and check that the mesh can be re-loaded and viewed correctly from the menu.

**Comment by Tuncay-Ayhan** [comment]
> I have come to enjoy staying current with FW changes and trying new things. However, I've come to hate having to go through manual mesh generation every time I flash a new FW and reset/re-save settings. I've noticed things like LCD Beeper setting, UI settings, etc appear to be retained after resetting EEPROM after FW flash. 
> 
> 
> 
> Would it be possible to have an option to retain mesh after EEPROM reset?

Unfortunately "Manual Mesh" is saved on eeprom. Jacobs explained that a short while back. The UBL mesh is saved on a dedicated slot. So it doesn't get wiped after a reset/reinstall of firmware.

**Reply by embraceentropy** [reply]
What is the difference between UBL with no probe vs. manual mesh?

**Comment by Tuncay-Ayhan** [comment]
Sorry, should be more specific. I ment ubl with bl touch. Read this about
the no-probe. It’s in the Wiki.

https://github.com/Jyers/Marlin/wiki/Configuration-Details#ubl



Op za 22 mei 2021 om 15:55 schreef embraceentropy ***@***.***>

**Reply by embraceentropy** [reply]
Thanks, didn't think to check there. I just compiled a UBL-No Probe version and am going to try it tonight
